pysersic/utils/results.py

```diff
@@ -1,43 +1,341 @@
+        out = 'results object for pysersic fit\n'
+        out+= f"\t contains SVI results {hasattr(self,'svi_results')}\n"
+        out+= f"\t contains sampling results: {hasattr(self,'sampling_results')}\n"
+        return out 
+
+
+
+
+    def add_prior(self,prior):
+        """add the prior object to the result object once created
+
+        Parameters
+        ----------
+        prior : pysersic.priors.BasePrior
+            created prior object
+        """
+        self.prior = prior 
+    def injest_data(self, 
+                sampler: Optional[numpyro.infer.mcmc.MCMC] =  None, 
+                svi_res_dict: Optional[dict] =  None,
+                purge_extra: Optional[bool] = True,
+                rkey: Optional[jax.random.PRNGKey] = random.PRNGKey(5)
+        ) -> pandas.DataFrame:
+        """Method to injest data from optimized SVI model or results of sampling. 
+        When sampling data is input, sets the class attribute `sampling_results`, while for an SVI run, saves `svi_results.`
+        A PysersicResults object can have at most data for one SVI and one sampling run. Both class atttributes are 
+        instances of arviz.InferenceData objects.
+
+        Parameters
+        ----------
+        sampler : Optional[numpyro.infer.mcmc.MCMC], optional
+            numpyro sampler containing results
+        svi_res_dict : Optional[dict], optional
+            Dictionary containing 'guide', 'model' and 'svi_result' specifying a trained SVI model
+        purge_extra : Optional[bool], optional
+            Whether to purge variables containing 'auto', 'base' or 'unwrapped' often used in reparamaterization, by default True
+        rkey : Optional[jax.random.PRNGKey], optional
+            PRNG key to use, by default jax.random.PRNGKey(5)
+
+        Returns
+        -------
+        pandas.DataFrame
+            ArviZ Summary of results
+
+        Raises
+        ------
+        AssertionError
+            Must supply one of sampler or svi_dict
+        """
+
+        if sampler is None and (svi_res_dict is None):
+            raise AssertionError("Must svi results dictionary or sampled sampler")
+
+        elif sampler is not None:
+            self.results = az.from_numpyro(sampler)
+            self.results = self._parse_injested_data(self.results,purge_extra=purge_extra)
+            self.runtype = 'sampling'
+        else:
+            assert 'guide' in svi_res_dict.keys()
+            assert 'model' in svi_res_dict.keys()
+            assert 'svi_result' in svi_res_dict.keys()
+
+            post_raw = svi_res_dict['guide'].sample_posterior(rkey, svi_res_dict['svi_result'].params, sample_shape = ((1000,)))
+            #Convert to arviz
+            post_dict = {}
+            for key in post_raw:
+                post_dict[key] = post_raw[key][jnp.newaxis,]
+            self.results = az.from_dict(post_dict)
+            self.results = self._parse_injested_data(self.results,purge_extra=purge_extra)
```

Comment:
I think better to call this attribute `self.posterior` or `self.idata` rather than `self.results`

Reply:
agreed, done!